// viem/_esm/experimental/index.js
// biome-ignore lint/performance/noBarrelFile: entrypoint module
export { getCapabilities, } from './eip5792/actions/getCapabilities.js';
export { sendCalls, } from './eip5792/actions/sendCalls.js';
export { getCallsStatus, } from './eip5792/actions/getCallsStatus.js';
export { showCallsStatus, } from './eip5792/actions/showCallsStatus.js';
export { WaitForCallsStatusTimeoutError, waitForCallsStatus, } from './eip5792/actions/waitForCallsStatus.js';
export { writeContracts, } from './eip5792/actions/writeContracts.js';
export { eip5792Actions, } from './eip5792/decorators/eip5792.js';
export { 
/** @deprecated This is no longer experimental – use `import { createWalletClient } from 'viem'` or `import { walletActions } from 'viem'` instead. */
walletActions as eip7702Actions, } from '../clients/decorators/wallet.js';
export { 
/** @deprecated This is no longer experimental – use `import { prepareAuthorization } from 'viem/actions'` instead. */
prepareAuthorization, } from '../actions/wallet/prepareAuthorization.js';
export { 
/** @deprecated This is no longer experimental – use `import { signAuthorization } from 'viem/actions'` instead. */
signAuthorization, } from '../actions/wallet/signAuthorization.js';
export {} from '../types/authorization.js';
export {} from '../types/rpc.js';
export { 
/** @deprecated This is no longer experimental – use `import { hashAuthorization } from 'viem/utils'` instead. */
hashAuthorization, } from '../utils/authorization/hashAuthorization.js';
export { 
/** @deprecated This is no longer experimental – use `import { recoverAuthorizationAddress } from 'viem/utils'` instead. */
recoverAuthorizationAddress, } from '../utils/authorization/recoverAuthorizationAddress.js';
export { 
/** @deprecated This is no longer experimental – use `import { serializeAuthorizationList } from 'viem/utils'` instead. */
serializeAuthorizationList, } from '../utils/authorization/serializeAuthorizationList.js';
export { 
/** @deprecated This is no longer experimental – use `import { verifyAuthorization } from 'viem/utils'` instead. */
verifyAuthorization, } from '../utils/authorization/verifyAuthorization.js';
export { grantPermissions, } from './erc7715/actions/grantPermissions.js';
export { erc7715Actions, } from './erc7715/decorators/erc7715.js';
export { erc7739Actions, } from './erc7739/decorators/erc7739.js';
export { erc7821Actions, } from './erc7821/decorators/erc7821.js';
